protocol/x/clob/keeper/grpc_stream_orderbook.go

```diff
@@ -18,3 +20,54 @@ func (k Keeper) StreamOrderbookUpdates(
 	<-ctx.Done()
 	return nil
 }
+
+// revertOptimisticFills sends OrderUpdates that effectively reverts the optimistic fills.
+func (k Keeper) revertOptimisticFills(ctx sdk.Context) {
+	// Send absolute fill amounts from local to the grpc stream.
+	// This must be sent out to account for checkState being discarded and deliverState being used.
+	localValidatorOperationsQueue, _ := k.MemClob.GetOperationsToReplay(ctx)
+	orderIdsFromLocalOpsQueue := fetchOrdersInvolvedInOpQueue(
+		localValidatorOperationsQueue,
+	)
+	allUpdates := types.NewOffchainUpdates()
+
+	for orderId := range orderIdsFromLocalOpsQueue {
+		orderbookUpdate := k.MemClob.GetOrderbookUpdatesForOrderUpdate(ctx, orderId)
+		allUpdates.Append(orderbookUpdate)
+	}
+	k.SendOrderbookUpdates(ctx, allUpdates)
+}
+
+// streamOnchainFills sends updates resulting from onchain fills.
+func (k Keeper) streamOnchainFills(ctx sdk.Context) {
+	// Get onchain stream events stored in transient store.
```

Comment:
nit: add a comment about how the optimistic fills are discarded from transient store if OE fails